packages/gatsby-recipes/src/joi-to-graphql/helpers/joi-to-graphql.js

```diff
@@ -112,10 +113,18 @@ internals.buildFields = fields => {
     const key = field.key
 
     if (field.schema._type === `object`) {
-      const Type = new GraphQLObjectType({
-        name: field.key.charAt(0).toUpperCase() + field.key.slice(1),
-        fields: internals.buildFields(field.schema._inner.children),
-      })
+      let Type
+      if (!field.schema._inner.children) {
+        // When there's an object with no children that means we've
+        // called Joi.object() and permit any key. As such let's specify
+        // a JSON GraphQL object type that's just as permissive.
+        Type = GraphQLJSONObject
```

Comment:
This is really neat! 👍  Feels intuitive to use `Joi.object()` and then be able to pass any key.